### PR TITLE
MODUSERBL-90: Upgrade to RMB v30.

### PR DESCRIPTION
**Pull request by bohdan-suprun**
https://issues.folio.org/browse/MODUSERBL-90 - Upgrade to RAML Module Builder 30.0.0

### CODE REVIEW

File: src/main/java/org/folio/rest/util/RestUtil.java [hunk 1]
```diff
@@ -63,8 +64,8 @@ private RestUtil() {
    * @return - async http response
    */
   public static Future<WrappedResponse> doRequest(HttpClient client, String url, HttpMethod method,
-                                                  CaseInsensitiveHeaders headers, String payload) {
-    Future<WrappedResponse> future = Future.future();
+                                                  MultiMap headers, String payload) {
```
Comment (marcjohnson-kint) [inline src/main/java/org/folio/rest/util/RestUtil.java:67]: Please fix the line continuation indentation whilst this line is being edited

File: src/test/java/org/folio/rest/GeneratePasswordRestLinkTest.java [hunk 2]
```diff
@@ -92,6 +92,7 @@ public static void setUp(TestContext context) {
       .setContentType(ContentType.JSON)
       .setBaseUri("http://localhost:" + port)
       .addHeader(RestVerticle.OKAPI_HEADER_TENANT, TENANT)
+      .addHeader(RestVerticle.OKAPI_HEADER_TOKEN, "test_token")
```
Comment (marcjohnson-kint) [inline src/test/java/org/folio/rest/GeneratePasswordRestLinkTest.java:95]: Why is this needed?

File: src/test/java/org/folio/rest/TestUtil.java [hunk 3]
```diff
@@ -86,9 +87,9 @@ public JsonObject getJson() {
           System.out.println(String.format(
                   "Got new WrappedResponse object with values code %s, body %s",
                   wr.getCode(), wr.getBody()));
-          if(!future.isComplete() && !future.failed()) {
+          if(!promise.future().isComplete() && !promise.future().failed()) {
```
Comment (marcjohnson-kint) [inline src/test/java/org/folio/rest/TestUtil.java:90]: Could `promise.future()` be extracted to a variable. Is it guaranteed that the future will be the same each time this is invoked?

File: src/test/java/org/folio/rest/GeneratePasswordRestLinkTest.java (hunk 2 above)
Comment (bohdan-suprun) [reply src/test/java/org/folio/rest/GeneratePasswordRestLinkTest.java:95]: I got NPE when we try to copy received headers in `org.folio.service.PasswordResetLinkServiceImpl#sendPasswordRestLink` endpoint  when calling `org.folio.rest.util.OkapiConnectionParams#buildHeaders`.

Probably `io.vertx.core.http.CaseInsensitiveHeaders` implementation changed and it is not allow nulls anymore

Comment (marcjohnson-kint) [reply src/test/java/org/folio/rest/GeneratePasswordRestLinkTest.java:95]: > I got NPE when we try to copy received headers in `org.folio.service.PasswordResetLinkServiceImpl#sendPasswordRestLink` endpoint when calling `org.folio.rest.util.OkapiConnectionParams#buildHeaders`.

Thanks for that explanation. Please could a comment be added to explain that in the code.

Also, if not too much work, please could a more representative token be used (like in mod-circulation)

> Probably `io.vertx.core.http.CaseInsensitiveHeaders` implementation changed and it is not allow nulls anymore

Hasn't that been replaced with `caseInsensitiveMultiMap` everywhere?

Comment (bohdan-suprun) [reply src/test/java/org/folio/rest/GeneratePasswordRestLinkTest.java:95]: Yes, it has. But this method internally uses that deprecated class.

File: src/test/java/org/folio/rest/TestUtil.java (hunk 3 above)
Comment (bohdan-suprun) [reply src/test/java/org/folio/rest/TestUtil.java]: Extracted to variable.

>  Is it guaranteed that the future will be the same each time this is invoked?

I think it is.

File: src/test/java/org/folio/rest/GeneratePasswordRestLinkTest.java (hunk 2 above)
Comment (bohdan-suprun) [reply src/test/java/org/folio/rest/GeneratePasswordRestLinkTest.java]: > Also, if not too much work, please could a more representative token be used (like in mod-circulation).

Done.
